src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -146,6 +146,57 @@ internal static MsalTokenResponse CreateFromiOSBrokerResponse(Dictionary<string,
             return response;
         }
 
+        internal static MsalTokenResponse CreateFromAppProviderResponse(ExternalTokenResult externalTokenResponse)
+        {
+            ValidateExternalTokenResult(externalTokenResponse);
+
+            var response = new MsalTokenResponse
+            {
+                AccessToken = externalTokenResponse.RawAccessToken,
+                RefreshToken = null,
+                IdToken = null,
+                TokenType = BrokerResponseConst.Bearer,
+                ExpiresIn = externalTokenResponse.ExpiresInSeconds,
+                ClientInfo = null,
+                TokenSource = TokenSource.IdentityProvider,
+                TenantId = externalTokenResponse.TenantId
+            };
+
+            if (externalTokenResponse.RefreshInSeconds.HasValue)
+            {
+                response.RefreshIn = externalTokenResponse.RefreshInSeconds.Value;
+            }
+            else
+            {
+                response.RefreshIn = response.ExpiresIn;
```

Comment:
response.RefreshIn is of type `long?` so it can be null. This block can be ` response.RefreshIn = externalTokenResponse.RefreshInSeconds`